Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -679,6 +679,28 @@ public protocol PurchasesType: AnyObject {
 
     #if os(iOS) || VISION_OS
 
+    /**
+     * Displays a sheet that enables users to redeem subscription offer codes that you generated in App Store Connect.
+     *
+     * - Important: Even though the docs in `SKPaymentQueue.presentCodeRedemptionSheet`
+     * say that it's available on Catalyst 14.0, there is a note:
+     * "`This function doesn’t affect Mac apps built with Mac Catalyst.`"
+     * when, in fact, it crashes when called both from Catalyst and also when running as "Designed for iPad".
+     * This is why RevenueCat's SDK makes it unavailable in mac catalyst.
+     */
```

Comment:
Does this mean calling this function of the SDK has been crashing on Catalyst and iOS apps running on macOS?

Reply:
For Catalyst, nope! Luckily the old version of `presentCodeRedemptionSheet()` was marked as unavailable on Catalyst (see the definition in the current main branch [here](https://github.com/RevenueCat/purchases-ios/blob/95aaf748887a6f5fdef1e45e3f9b2da67b2ad97c/Sources/Purchasing/Purchases/PurchasesType.swift#L753).

For "Designed for iPad" on macOS, I haven't tested it myself, but I wouldn't be surprised if the existing SDK function is crashing, since this PR introduces the `isiOSAppOnMac` check for the first time in the code redemption sheet flow